coaster/sqlalchemy.py

```diff
@@ -166,6 +166,26 @@ def __init__(self, *args, **kw):
         if not self.name:
             self.make_name()
 
+    @classmethod
+    def get(cls, name):
+        """Get an instance matching the name"""
+        return cls.query.filter_by(name=name).one_or_none()
+
+    @classmethod
+    def upsert(cls, name, **fields):
+        """Insert or update an instance"""
+        instance = cls.get(name)
+        if instance:
+            for f in fields:
+                if hasattr(instance, f):
+                    setattr(instance, f, fields[f])
+                else:
+                    raise TypeError("'{arg}' is an invalid keyword argument for {instance_type}".format(arg=f, instance_type=instance.__class__.__name__))
```

Comment:
While SQLAlchemy places an instance with dirty fields in [`session.dirty`](http://docs.sqlalchemy.org/en/rel_0_9/orm/session_api.html#sqlalchemy.orm.session.Session.dirty), it talks to the database only if there is a net change. So, this shouldn't be an issue.